autometa/taxonomy/ncbi.py

```diff
@@ -366,47 +372,64 @@ def is_common_ancestor(self, taxid_A, taxid_B):
-        level=logger.DEBUG,
-    )
-    parser = argparse.ArgumentParser(description="Autometa NCBI utilities class")
-    parser.add_argument("ncbi", help="</path/to/ncbi/database/directory>")
-    parser.add_argument(
-        "--query-taxid",
-        help="query taxid to test NCBI class functionality",
-        default=1222,
-        type=int,
-    )
-    parser.add_argument(
-        "--verbose", help="add verbosity", action="store_true", default=True
-    )
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f"{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n"
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f"{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n"
-        f"{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n"
-        f"{args.query_taxid} is_common_ancestor {query_taxid_parent}: "
-        f"{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n"
-    )
+        lineage_a_taxids = {ancestor.get("taxid") for ancestor in self.lineage(taxid_A)}
+        lineage_b_taxids = {ancestor.get("taxid") for ancestor in self.lineage(taxid_B)}
+        common_ancestor = lineage_b_taxids.intersection(lineage_a_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given `taxid` is a positive integer or not as well as if it is present in nodes.dmp or names.dmp
+        In case the taxid provided is in the form of a string the function will cast the string as an integer
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in NCBI taxonomy databases - nodes.dmp, names.dmp or merged.dmp
+
+        Returns
+        -------
+        int
+            `taxid` if the `taxid` is a positive integer and present in either nodes.dmp or names.dmp or
+            new_taxid which is equal to the secondary `taxid` present in merged.dmp
+
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
+        """
+        # This checks if an integer has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            invalid_chars = set(string.punctuation + string.ascii_letters)
+            invalid_chars.discard(".")
+            if {char for char in taxid if char in invalid_chars} or taxid.count(
+                "."
+            ) > 1:
+                raise ValueError(f"taxid contains invalid character(s)! Given: {taxid}")
+            taxid = float(taxid)
+        # a boolean check is needed as something can be a boolean but not necessarily a str, eg. TRUE/FALSE
```

Comment:
This comment is not strictly true. When cast as an `int` or `float`, `True` evaluates to `1` and `False` evaluates to `0`. I suggest:

```suggestion
        # a boolean check is needed as they will evaluate silently to 0 or 1 when cast as ints.
```